include/drivers/can.h

```diff
@@ -508,28 +560,208 @@ static inline void z_impl_can_detach(const struct device *dev, int filter_id)
+ */
+int can_calc_timing_data(const struct device *dev, struct can_timing *res,
+			 uint32_t bitrate, uint16_t sample_pnt);
+#endif
+
+/**
+ * @brief Fill in the prescaler value for a given bitrate and timing
+ *
+ * Fill the prescaler value in the timing struct.
+ * sjw, prop_seg, phase_seg1 and phase_seg2 must be given.
+ *
+ * @param dev     Pointer to the device structure for the driver instance.
+ * @param timing  Result is written into the can_timing struct provided.
+ * @param bitrate Target bitrate.
+ *
+ * @retval bitrate error
+ */
+int can_calc_prescaler(const struct device *dev, struct can_timing *timing,
+		       uint32_t bitrate);
+
 /**
  * @brief Configure operation of a host controller.
  *
+ * The timing structs can either be valid timings or NULL. If NULL is passed,
+ * the bus timing does not change. timing_data is only relevant for CAN-FD
+ * If the controller does not support CAN-FD or the FD mode is not enabled,
+ * this parameter is ignored.
+ *
  * @param dev Pointer to the device structure for the driver instance.
- * @param mode Operation mode
- * @param bitrate bus-speed in Baud/s
+ * @param mode   Operation mode
+ * @param timing Bus timings
+ * @param timing Bus timings for the dataphase (CAN-FD only)
  *
  * @retval 0 If successful.
  * @retval -EIO General input / output error, failed to configure device.
  */
 __syscall int can_configure(const struct device *dev, enum can_mode mode,
-			    uint32_t bitrate);
+			    const struct can_timing *timing,
+			    const struct can_timing *timing_data);
 
-static inline int z_impl_can_configure(const struct device *dev,
-				       enum can_mode mode,
-				       uint32_t bitrate)
+static inline int z_impl_can_configure(const struct device *dev, enum can_mode mode,
+				       const struct can_timing *timing,
+				       const struct can_timing *timing_data)
 {
 	const struct can_driver_api *api =
 		(const struct can_driver_api *)dev->api;
 
-	return api->configure(dev, mode, bitrate);
+	return api->configure(dev, mode, timing, timing_data);
+}
+
+/**
+ * @brief Set the controller to the given mode
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param mode   Operation mode
+ *
+ * @retval 0 If successful.
+ * @retval -EIO General input / output error, failed to configure device.
+ */
+static inline int can_set_mode(const struct device *dev, enum can_mode mode)
+{
+	return can_configure(dev, mode, NULL, NULL);
+}
+
+/**
+ * @brief Set the bitrate of the controller
+ *
+ * The sample point is set to the CiA DS 301 reccommended value of 87.5%
+ *
+ * @param dev     Pointer to the device structure for the driver instance.
+ * @param bitrate Desired bitrate
+ *
+ * @retval 0 If successful.
+ * @retval -EINVAL bitrate cannot be reached.
+ * @retval -EIO General input / output error, failed to configure device.
+ */
+static inline int can_set_bitrate(struct device *dev, uint32_t bitrate)
```

Comment:
```suggestion
static inline int can_set_bitrate(const struct device *dev, uint32_t bitrate)
```
(same for the functions below)